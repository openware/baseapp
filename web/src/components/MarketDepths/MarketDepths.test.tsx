import { render } from '@testing-library/react';
import React from 'react';
import { TestComponentWrapper } from 'src/lib/test';
import { MarketDepths, MarketDepthsProps } from '.';

class ResizeObserver {
    observe() {
        // do nothing
    }
    unobserve() {
        // do nothing
    }
    disconnect() {
        // do nothing
    }
}

window.ResizeObserver = ResizeObserver;

const defaultProps: MarketDepthsProps = {
    data: [
        { x: '', bid: 4500, amt: 1 },
        { x: '', bid: 3500, amt: 2 },
        { x: '$ 4.000', bid: 2359, amt: 3 },
        { x: '', bid: 2256 },
        { x: '', bid: 2222 },
        { x: '', bid: 2190 },
        { x: '$ 3.526', bid: 2014 },
        { x: '', bid: 2003 },
        { x: '', bid: 1970 },
        { x: '', bid: 1905 },
        { x: '', bid: 1900 },
        { x: '', bid: 1890 },
        { x: '', bid: 1670 },
        { x: '$ 3.150', bid: 560 },
        { x: '', bid: 480 },
        { x: '', bid: 470 },
        { x: '', bid: 310 },
        { x: '', bid: 234 },
        { x: '', bid: 100, ask: 110 },
        { x: '', ask: 800 },
        { x: '', ask: 940 },
        { x: '', ask: 967 },
        { x: '', ask: 999 },
        { x: '$ 3.150', ask: 1100 },
        { x: '', ask: 1130 },
        { x: '', ask: 1800 },
        { x: '', ask: 2100 },
        { x: '', ask: 2600 },
        { x: '$ 3.526', ask: 3800 },
        { x: '', ask: 4100 },
        { x: '', ask: 4300 },
        { x: '', ask: 4370 },
        { x: '', ask: 4500 },
        { x: '', ask: 4680 },
        { x: '', ask: 4790 },
        { x: '$ 4.000', ask: 4800 },
        { x: '', ask: 5100 },
        { x: '', ask: 5250 },
    ],
};

const renderComponent = (props: Partial<MarketDepthsProps> = {}) =>
    render(
        <TestComponentWrapper>
            <MarketDepths {...{ ...defaultProps, ...props }} />
        </TestComponentWrapper>,
    );

describe('CryptoIcon', () => {
    it('should render', () => {
        expect(renderComponent().container).toMatchSnapshot();
    });

    it('should have correct className', () => {
        expect(renderComponent().container.querySelector('.cr-market-depths')).toBeInTheDocument();
    });

    it('always renders a market depth', () => {
        expect(renderComponent().container).toBeInTheDocument();
    });
});
