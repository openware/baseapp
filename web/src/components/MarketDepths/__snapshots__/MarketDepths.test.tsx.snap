// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoIcon should render 1`] = `
<div
  className="cr-market-depths"
>
  <ResponsiveContainer
    debounce={0}
    height="100%"
    width="100%"
  >
    <AreaChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Object {
            "amt": 1,
            "bid": 4500,
            "x": "",
          },
          Object {
            "amt": 2,
            "bid": 3500,
            "x": "",
          },
          Object {
            "amt": 3,
            "bid": 2359,
            "x": "$ 4.000",
          },
          Object {
            "bid": 2256,
            "x": "",
          },
          Object {
            "bid": 2222,
            "x": "",
          },
          Object {
            "bid": 2190,
            "x": "",
          },
          Object {
            "bid": 2014,
            "x": "$ 3.526",
          },
          Object {
            "bid": 2003,
            "x": "",
          },
          Object {
            "bid": 1970,
            "x": "",
          },
          Object {
            "bid": 1905,
            "x": "",
          },
          Object {
            "bid": 1900,
            "x": "",
          },
          Object {
            "bid": 1890,
            "x": "",
          },
          Object {
            "bid": 1670,
            "x": "",
          },
          Object {
            "bid": 560,
            "x": "$ 3.150",
          },
          Object {
            "bid": 480,
            "x": "",
          },
          Object {
            "bid": 470,
            "x": "",
          },
          Object {
            "bid": 310,
            "x": "",
          },
          Object {
            "bid": 234,
            "x": "",
          },
          Object {
            "ask": 110,
            "bid": 100,
            "x": "",
          },
          Object {
            "ask": 800,
            "x": "",
          },
          Object {
            "ask": 940,
            "x": "",
          },
          Object {
            "ask": 967,
            "x": "",
          },
          Object {
            "ask": 999,
            "x": "",
          },
          Object {
            "ask": 1100,
            "x": "$ 3.150",
          },
          Object {
            "ask": 1130,
            "x": "",
          },
          Object {
            "ask": 1800,
            "x": "",
          },
          Object {
            "ask": 2100,
            "x": "",
          },
          Object {
            "ask": 2600,
            "x": "",
          },
          Object {
            "ask": 3800,
            "x": "$ 3.526",
          },
          Object {
            "ask": 4100,
            "x": "",
          },
          Object {
            "ask": 4300,
            "x": "",
          },
          Object {
            "ask": 4370,
            "x": "",
          },
          Object {
            "ask": 4500,
            "x": "",
          },
          Object {
            "ask": 4680,
            "x": "",
          },
          Object {
            "ask": 4790,
            "x": "",
          },
          Object {
            "ask": 4800,
            "x": "$ 4.000",
          },
          Object {
            "ask": 5100,
            "x": "",
          },
          Object {
            "ask": 5250,
            "x": "",
          },
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 20,
          "left": 0,
          "right": 30,
          "top": 20,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
    >
      <defs>
        <linearGradient
          id="bidChartColor"
          x1="0"
          x2="0"
          y1="0"
          y2="1"
        >
          <stop
            offset="5%"
            stopColor=""
            stopOpacity={1}
          />
          <stop
            offset="95%"
            stopColor=""
            stopOpacity={0}
          />
        </linearGradient>
        <linearGradient
          id="askChartColor"
          x1="0"
          x2="0"
          y1="0"
          y2="1"
        >
          <stop
            offset="5%"
            stopColor=""
            stopOpacity={1}
          />
          <stop
            offset="95%"
            stopColor=""
            stopOpacity={0}
          />
        </linearGradient>
        <linearGradient
          id="fillGrid"
          x1="0"
          x2="0"
          y1="0"
          y2="1"
        >
          <stop
            offset="5%"
            stopColor=""
            stopOpacity={0}
          />
          <stop
            offset="95%"
            stopColor=""
            stopOpacity={1}
          />
        </linearGradient>
      </defs>
      <CartesianGrid
        fill="url(#fillGrid)"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke=""
        strokeDasharray="1 1"
        vertical={true}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="ask"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={30}
        hide={false}
        interval="preserveStartEnd"
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        stroke=""
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="bid"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={0}
        hide={false}
        interval="preserveStartEnd"
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        stroke=""
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        content={
          <CustomTooltip
            external={
              Array [
                Object {
                  "amt": 1,
                  "bid": 4500,
                  "x": "",
                },
                Object {
                  "amt": 2,
                  "bid": 3500,
                  "x": "",
                },
                Object {
                  "amt": 3,
                  "bid": 2359,
                  "x": "$ 4.000",
                },
                Object {
                  "bid": 2256,
                  "x": "",
                },
                Object {
                  "bid": 2222,
                  "x": "",
                },
                Object {
                  "bid": 2190,
                  "x": "",
                },
                Object {
                  "bid": 2014,
                  "x": "$ 3.526",
                },
                Object {
                  "bid": 2003,
                  "x": "",
                },
                Object {
                  "bid": 1970,
                  "x": "",
                },
                Object {
                  "bid": 1905,
                  "x": "",
                },
                Object {
                  "bid": 1900,
                  "x": "",
                },
                Object {
                  "bid": 1890,
                  "x": "",
                },
                Object {
                  "bid": 1670,
                  "x": "",
                },
                Object {
                  "bid": 560,
                  "x": "$ 3.150",
                },
                Object {
                  "bid": 480,
                  "x": "",
                },
                Object {
                  "bid": 470,
                  "x": "",
                },
                Object {
                  "bid": 310,
                  "x": "",
                },
                Object {
                  "bid": 234,
                  "x": "",
                },
                Object {
                  "ask": 110,
                  "bid": 100,
                  "x": "",
                },
                Object {
                  "ask": 800,
                  "x": "",
                },
                Object {
                  "ask": 940,
                  "x": "",
                },
                Object {
                  "ask": 967,
                  "x": "",
                },
                Object {
                  "ask": 999,
                  "x": "",
                },
                Object {
                  "ask": 1100,
                  "x": "$ 3.150",
                },
                Object {
                  "ask": 1130,
                  "x": "",
                },
                Object {
                  "ask": 1800,
                  "x": "",
                },
                Object {
                  "ask": 2100,
                  "x": "",
                },
                Object {
                  "ask": 2600,
                  "x": "",
                },
                Object {
                  "ask": 3800,
                  "x": "$ 3.526",
                },
                Object {
                  "ask": 4100,
                  "x": "",
                },
                Object {
                  "ask": 4300,
                  "x": "",
                },
                Object {
                  "ask": 4370,
                  "x": "",
                },
                Object {
                  "ask": 4500,
                  "x": "",
                },
                Object {
                  "ask": 4680,
                  "x": "",
                },
                Object {
                  "ask": 4790,
                  "x": "",
                },
                Object {
                  "ask": 4800,
                  "x": "$ 4.000",
                },
                Object {
                  "ask": 5100,
                  "x": "",
                },
                Object {
                  "ask": 5250,
                  "x": "",
                },
              ]
            }
          />
        }
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        separator=" : "
        useTranslate3d={false}
        viewBox={
          Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Area
        activeDot={
          Object {
            "fill": "rgba(91,165,132,1)",
            "r": 4,
            "stroke": "rgba(31,42,52,1)",
            "strokeWidth": 1,
          }
        }
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="bid"
        dot={false}
        fill="url(#bidChartColor)"
        fillOpacity={0.6}
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke=""
        type="step"
        xAxisId={0}
        yAxisId={0}
      />
      <Area
        activeDot={
          Object {
            "fill": "rgba(91,165,132,1)",
            "r": 4,
            "stroke": "rgba(31,42,52,1)",
            "strokeWidth": 1,
          }
        }
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="ask"
        dot={false}
        fill="url(#askChartColor)"
        fillOpacity={0.6}
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke=""
        type="step"
        xAxisId={0}
        yAxisId={0}
      />
    </AreaChart>
  </ResponsiveContainer>
</div>
`;
